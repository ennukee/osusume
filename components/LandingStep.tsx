import React from 'react';
import styled from 'styled-components';

interface StepProps {
  number: number;
  title: string;
  children: string;
}
export const LandingStep: React.FC<StepProps> = ({ number, title, children }) => {
  return (
    <Container>
      <StepBackLine />
      <StepNumber>{number}</StepNumber>
      <StepTitle>{title}</StepTitle>
      <StepDescription>{children}</StepDescription>
    </Container>
  );
};

const Container = styled.div`
  position: relative;
  display: flex;
  flex: 1 1 0px;
  flex-direction: column;
  align-items: center;
  margin: 30px;
`;

const StepBackLine = styled.div`
  position: absolute;
  height: 1px;
  width: 100%;
  top: 25px;
  background-color: black;
`;

const StepNumber = styled.div`
  padding: 0 10px;
  background-color: white;
  font-weight: 900;
  font-size: 40px;
  z-index: 2;
  font-family: 'Quicksand', sans-serif;
`;

const StepTitle = styled.div`
  font-size: 24px;
  margin-top: -5px;
`;

const StepDescription = styled.div`
  font-size: 14px;
  text-align: center;
  margin-top: 15px;
`;
