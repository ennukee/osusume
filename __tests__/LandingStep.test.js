import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LandingStep } from '../components/LandingStep.tsx';

describe('components/LandingStep', () => {
  let container;
  beforeEach(() => {
    ({ container } = render(
      <LandingStep number={1} title="Title">
        Children
      </LandingStep>,
    ));
  });
  describe('props', () => {
    it('matches snapshot', () => {
      expect(container).toMatchSnapshot();
    });
  });
});
