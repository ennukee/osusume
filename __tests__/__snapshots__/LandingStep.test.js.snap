// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LandingStep props matches snapshot 1`] = `
<div>
  <div
    class="LandingStep__Container-vcdptp-0 gqyyxB"
  >
    <div
      class="LandingStep__StepBackLine-vcdptp-1 hHJrnk"
    />
    <div
      class="LandingStep__StepNumber-vcdptp-2 eTpyhT"
    >
      1
    </div>
    <div
      class="LandingStep__StepTitle-vcdptp-3 lnfRyO"
    >
      Title
    </div>
    <div
      class="LandingStep__StepDescription-vcdptp-4 cZRtMj"
    >
      Children
    </div>
  </div>
</div>
`;
